app/controllers/cloud_volume_controller.rb

```diff
@@ -129,6 +129,22 @@ def cloud_volume_form_fields
     }
   end
 
+  def storage_manager_form_fields
+    assert_privileges("cloud_volume_new")
+    ems = find_by_id_filtered(ExtManagementSystem, params[:id])
+
+    form_fields = { :emstype => ems.emstype }
+
+    case ems.type
+    when "ManageIQ::Providers::Amazon::StorageManager::Ebs"
+      form_fields[:availability_zone_choices] = ems.availability_zones.each_with_object([]) { |az, choices| choices << az.name }.sort
+    when "ManageIQ::Providers::StorageManager::CinderManager"
+      form_fields[:cloud_tenant_choices] = ems.parent_manager.cloud_tenants.each_with_object({}) { |tenant, choices| choices[tenant.name] = tenant.id }
+    end
```

Comment:
This seems like it should work!

There is a difference in the way Amazon and OpenStack managers work. First, [here](https://github.com/ManageIQ/manageiq-ui-classic/pull/517/files#diff-6e783b8caed5140ba7bdfad51acb5f6cR259) we are retrieving managers that have `supports_block_storage` set.

For Amazon, this is set in [EBS block storage manager](https://github.com/ManageIQ/manageiq-providers-amazon/blob/master/app/models/manageiq/providers/amazon/storage_manager/ebs.rb#L9). For OpenStack in the [Cinder manager](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/storage_manager/cinder_manager.rb#L11).

EBS delegates retrieval of availability zones to the parent manager because it knows its parent manager is the Amazon cloud manager. On the other hand, Cinder manager is more generic because it can be used by multiple different cloud managers (not currently, but in theory). Therefore, there was no delegation to the parent made so far (and I didn't want to add it just for the sake of using it here).

That being said, as you've seen while preparing the API call, explicitly including the `parent_manager` in case of EBS also works so I think we can use them in both cases to not cause confusion.